opds_import.py

```diff
@@ -1911,6 +1911,17 @@ def identifier_needs_import(self, identifier, last_updated_remote):
                 )
                 return True
 
+        # The imported title might be known already but not in the collection we are importing,
+        # check for that condition
+        has_pool = LicensePool.exists(self._db, identifier, self.collection_id, self.importer.data_source_name)
+        if not has_pool:
+            # The collection doesnt have this title. Importing it will create a LicensePool
+            self.log.info(
+                "Counting %s as new because it is not in the collection.",
+                identifier
+            )
+            return True
+
```

Comment:
This makes sense; in fact I think this check should go first. If there's no LicensePool for this book in the collection then there's no need to check for an import record or an import failure -- if it happened, it clearly didn't work.